crates/kitsune_p2p/fetch/src/pool.rs

```diff
@@ -90,26 +90,37 @@ impl FetchPoolConfig for FetchPoolConfigBitwiseOr {
 }
 
 /// The actual inner state of the FetchPool, from which items can be obtained
-#[derive(Debug, Default)]
-pub struct State {
+#[derive(Debug)]
+pub struct FetchPoolState {
+    config: FetchConfig,
```

Comment:
Why is the config moving to the state? It should be immutable from what I remember of working on this code so it seems a bit misleading to bundle it into the state

Reply:
@ThetaSinner yes, this was just a convenience, because without this you have to pass the config in every time. I think you're right that it's better to keep the state purely mutable.